### DISCUSSION TITLE
Can I use Chipyard FPGA flow with the Xilinx ZYNQ ZC706 board?

### DISCUSSION THREAD

**Discussion by sammy17** [ucb-bar/chipyard#1477] [Q&A]
Can I use Chipyard FPGA flow with the Xilinx ZYNQ ZC706 board? 
I saw that there are several FPGAs supported as shown in https://github.com/ucb-bar/chipyard/blob/main/fpga/Makefile
I just wanted to check whether it's possible to use ZC706 board for the FPGA flow and if so, what are the changes that I need to do for it? 

Thank you in advance!

**Comment by milovanovic** [comment]
You can use Chipyard's FPGA prototyping flow with any Xilinx evaluation board, including the [ZC706](https://www.xilinx.com/products/boards-and-kits/ek-z7-zc706-g.html). However, you would need to manually add the support for it. In order to do so, you basically have to provide Configs, Harness and Binders in Scala just as done [here](https://github.com/ucb-bar/chipyard/tree/main/fpga/src/main/scala) for the supported development boards, and add the appropriate XDC constraints and Vivado TCL scripts in the [FPGA Shells](https://github.com/chipsalliance/rocket-chip-fpga-shells).

Bare in mind that currently only [Arty A7](https://digilent.com/reference/programmable-logic/arty-a7/start) and [VCU118](https://www.xilinx.com/products/boards-and-kits/vcu118.html) boards are supported by Chipyard (with some limited support for [VC707](https://www.xilinx.com/products/boards-and-kits/ek-v7-vc707-g.html), too).

It might be helpful for you to take a look at the [Prototyping Flow](https://chipyard.readthedocs.io/en/stable/Prototyping) documentation section.
